doc/tutorial/using_gpu.txt

```diff
@@ -532,23 +192,61 @@ What Can be Accelerated on the GPU
+  to keep 30 processors busy.
+* Indexing, dimension-shuffling and constant-time reshaping will be equally fast
+  on GPU as on CPU.
+* Summation over rows/columns of tensors can be a little slower on the
+  GPU than on the CPU.
+* Copying of large quantities of data to and from a device is relatively slow,
+  and often cancels most of the advantage of one or two accelerated functions
+  on that data. Getting GPU performance largely hinges on making data transfer
+  to the device pay off.
+
+The backend supports all regular theano data types (float32, float64,
+int, ...), however GPU support varies and some units can't deal with
 double (float64) or small (less than 32 bits like int16) data types.
 You will get an error at compile time or runtime if this is the case.
 
-By default all inputs will get transferred to GPU.  You can prevent an
+By default all inputs will get transferred to GPU. You can prevent an
 input from getting transferred by setting its tag.target attribute to
 'cpu'.
 
 Complex support is untested and most likely completely broken.
 
-In general, large operations like matrix multiplication, or
-element-wise operations with large inputs, will be significatly
-faster.
+Tips for Improving Performance on GPU
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+* Consider adding ``floatX=float32`` (or the type you are using) to your
+  ``.theanorc`` file if you plan to do a lot of GPU work.
+* The GPU backend supports *float64* variables, but they are still slower
+  to compute than *float32*. The more *float32*, the better GPU performance
+  you will get. 
+* Prefer constructors like ``matrix``, ``vector`` and ``scalar`` to
+  ``dmatrix``, ``dvector`` and ``dscalar`` because the former will give
+  you *float32* variables and ignore the type given to ``floatX``.
+* Minimize transfers to the GPU device by using ``shared`` variables
+  to store frequently-accessed data (see :func:`shared()<shared.shared>`).
+  When using the GPU, tensor ``shared`` variables are stored on
+  the GPU by default to eliminate transfer time for GPU ops using those variables.
+* If you aren't happy with the performance you see, try running your script with
+  ``profile=True`` flag. This should print some timing information at program
+  termination. Is time being used sensibly?   If an op or Apply is
+  taking more time than its share, then if you know something about GPU
+  programming, have a look at how it's implemented in theano.gpuarray.
+  Check the line similar to *Spent Xs(X%) in cpu op, Xs(X%) in gpu op and
+  Xs(X%) in transfer op*. This can tell you if not enough of your graph is
+  on the GPU or if there is too much memory transfer.
+* To investigate whether all the Ops in the computational graph are
+  running on GPU, it is possible to debug or check your code by providing
+  a value to `assert_no_cpu_op` flag, i.e. `warn`, for warning, `raise` for
+  raising an error or `pdb` for putting a breakpoint in the computational
+  graph if there is a CPU Op.
```

Comment:
Unfortunately, that does nothing in the new back-end for the moment.
I'll make a PR for that.